test/plugins/FeatureInfoPlugin.test.js

```diff
@@ -42,11 +81,158 @@ describe('FeatureInfoPlugin', () => {
+				const coordinate = [11, 22];
+				const zoom = 5;
+				const resolution = 25;
+				const store = setup({
+					layers: {
+						active: [createDefaultLayer(layerId0)]
+					},
+					position: {
+						zoom: zoom
+					}
+				});
+				const instanceUnderTest = new FeatureInfoPlugin();
+
+				spyOn(mapService, 'calcResolution').withArgs(zoom, coordinate).and.returnValue(resolution);
+				spyOn(featureInfoService, 'get').withArgs(layerId0, coordinate, resolution).and.resolveTo(new FeatureInfoResult('content', 'title'));
+				await instanceUnderTest.register(store);
+
+				setClick({ coordinate: coordinate, screenCoordinate: [33, 44] });
+
+				setTimeout(() => {
+					expect(store.getState().featureInfo.current).toHaveSize(1);
+					expect(store.getState().featureInfo.current[0].content).toBe('content');
+					expect(store.getState().featureInfo.current[0].title).toBe('title');
+				});
+			});
+
+			it('adds FeatureInfo items taking layerPropeties\' label as title', async () => {
+				const layerId0 = 'id0';
+				const layerLabel0 = 'label0';
+				const coordinate = [11, 22];
+				const zoom = 5;
+				const resolution = 25;
+				const store = setup({
+					layers: {
+						active: [{ ...createDefaultLayer(layerId0), label: layerLabel0 }]
+					},
+					position: {
+						zoom: zoom
+					}
+				});
+				const instanceUnderTest = new FeatureInfoPlugin();
+
+				spyOn(mapService, 'calcResolution').withArgs(zoom, coordinate).and.returnValue(resolution);
+				spyOn(featureInfoService, 'get').withArgs(layerId0, coordinate, resolution).and.resolveTo(new FeatureInfoResult('content'));
+				await instanceUnderTest.register(store);
+
+				setClick({ coordinate: coordinate, screenCoordinate: [33, 44] });
+
+				setTimeout(() => {
+					expect(store.getState().featureInfo.current).toHaveSize(1);
+					expect(store.getState().featureInfo.current[0].content).toBe('content');
+					expect(store.getState().featureInfo.current[0].title).toBe(layerLabel0);
+				});
+			});
+
+			it('adds No FeatureInfo items when layer is invisible or hidden', async () => {
+				const layerId0 = 'id0';
+				const layerId1 = 'id1';
+				const coordinate = [11, 22];
+				const zoom = 5;
+				const resolution = 25;
+				const store = setup({
+					layers: {
+						active: [
+							{ ...createDefaultLayer(layerId0), constraints: { hidden: true } },
+							{ ...createDefaultLayer(layerId1), visible: false }
+						]
+					},
+					position: {
+						zoom: zoom
+					}
+				});
+				const instanceUnderTest = new FeatureInfoPlugin();
+
+				spyOn(mapService, 'calcResolution').withArgs(zoom, coordinate).and.returnValue(resolution);
+				const featureInfoServiceSpy = spyOn(featureInfoService, 'get');
+				await instanceUnderTest.register(store);
+
+				setClick({ coordinate: coordinate, screenCoordinate: [33, 44] });
+
+				expect(featureInfoServiceSpy).not.toHaveBeenCalled();
+			});
+
+			it('adds No FeatureInfo items when service returns no result', async () => {
```

Comment:
typo ...somehow -> 'No'

...if the capitalize was by intention, more a point for a style discussion later on: I used in my tests the pattern, to capitalize the negation , too. 
Should me make this a common rule? 
In this case: "adds NO FeatureInfo items when service returns NO result"

Reply:
good point